private/buf/bufgen/internal/bufgenv2/config.go

```diff
@@ -127,24 +136,75 @@ func newManagedConfig(logger *zap.Logger, externalConfig ExternalManagedConfigV2
 		// continue to validate this config
 	}
 	var disabledFuncs []disabledFunc
+	var fieldDisableFuncs []fieldDisableFunc
 	fileOptionGroupToOverrideFuncs := make(map[fileOptionGroup][]overrideFunc)
+	fieldOptionToOverrideFuncs := make(map[fieldOption][]fieldOverrideFunc)
 	for _, externalDisableConfig := range externalConfig.Disable {
-		if len(externalDisableConfig.FileOption) == 0 && len(externalDisableConfig.Module) == 0 && len(externalDisableConfig.Path) == 0 {
-			return nil, errors.New("must set one of file_option, module and path for a disable rule")
+		if len(externalDisableConfig.FileOption) == 0 &&
+			len(externalDisableConfig.FieldOption) == 0 &&
+			len(externalDisableConfig.Module) == 0 &&
+			len(externalDisableConfig.Path) == 0 &&
+			len(externalDisableConfig.Field) == 0 {
+			return nil, errors.New("must set one of file_option, field option, module, path and field for a disable rule")
+		}
+		if len(externalDisableConfig.FieldOption) > 0 && len(externalDisableConfig.FileOption) > 0 {
+			return nil, errors.New("only one of file_option and field_option can be specified in a disable rule")
 		}
+		if len(externalDisableConfig.FileOption) > 0 {
+			// this is a file option rule only
+			if len(externalDisableConfig.Field) > 0 {
+				return nil, errors.New("cannot specify both file option and field option")
+			}
+			disabledFunc, err := newDisabledFunc(externalDisableConfig)
+			if err != nil {
+				return nil, err
+			}
+			disabledFuncs = append(disabledFuncs, disabledFunc)
+			continue
+		}
+		if len(externalDisableConfig.FieldOption) > 0 || len(externalDisableConfig.Field) > 0 {
+			// this is a field option rule only
+			fieldDisableFunc, err := newFieldDisabledFunc(externalDisableConfig)
+			if err != nil {
+				return nil, err
+			}
+			fieldDisableFuncs = append(fieldDisableFuncs, fieldDisableFunc)
+			continue
+		}
+		// none of field_option, field and file_option is set. We disable both file options and field options.
```

Comment:
This is worthy of attention: if a disable rule is ambiguous on whether it disables field option or file option, disable both.